IPs/RSKIP178.md

```diff
@@ -0,0 +1,375 @@
+
+
+|RSKIP          | 178 |
+| :------------ |:-------------|
+|**Title**      |External Confirmation Hashrate|
+|**Created**    |SEP-2020 |
+|**Author**     |SDL |
+|**Purpose**    |Sec |
+|**Layer**      |Core |
+|**Complexity** |2 |
+|**Status**     |Draft |
+|**discussions-to**     | |
+
+
+# **Abstract**
+
+The Strong Fork-aware Merge-mining (SFAMM) protocol proposes a new security model suited for a merge-mined blockchain. If the blockchain implements the SFAMM protocol, and if the assumptions of the model are satisfied, the model assures that large double-spend attacks are prevented, by making attacks visible, attributable, requiring at least 51% Bitcoin miners collusion and, more importantly, requiring a bounded time, where the bound depends on the attacker hashrate but it is known in advance. The bound parameter can be tuned so that in case of attack a community can coordinate actions, such as arranging to manually invalidate a block, or perform a hard-fork, over side channels (such as social networks and forums) in order to deter the attack. 
+
+This RSKIP represents one of the changes to the RSK consensus that is required to achieve security under the SFAMM model.  
+
+
+
+
+## Motivation
+
+The only contribution to the cryptoeconomic security of an independent proof-of-work chain without subsidy is the existence of high transaction fees. In the case of sidechain 1:1 pegged to a primary chain that is protected by merge-mining, there is a high strategic alignment between the primary chain miners and merge-miners, so the honest majority and the fact that attacks to the secondary chain often become attributable are the two main assurances for the security for the sidechain. Nevertheless, RSK went further, and deployed the Armadillo system to provide fork-awareness. This subsystem creates a time window to trigger an emergency response by the community, in case of long reorg attempts.  Nevertheless, Armadillo requires a separate communication channel and the existence of at least one honest node monitoring the Bitcoin network. The SFAMM protocol and security model captures the benefits of Armadillo, makes Armadillo decentralized, and formalizes the protocol algorithms, assumptions and guarantees. The SFAMM protocol is ideal for RSK as it focuses not only on the hashrate capability of the attacker but also on the processing time that will be required for a successful attack.
+
+This RSKIP proposes a consensus change that enables RSK nodes to:
+
+- compute the active SHA256 hashrate and detect variations on the active hashrate that are compatible with an ongoing attack
+
+- Accumulate all the active SHA256 hashrate in the RSK blockchain, with independence of the amount of hashrate that is directly involved in merge-mining. This hashrate means more accumulated work, increasing the security against certain double-spend attacks. Also, the hashrate accumulated not only includes 100% of Bitcoin hashrate, but also hashrate from bitcoin forks.
```

Comment:
- Accumulate all the active SHA256 hashrate in the RSK blockchain, with independence of the amount of hashrate that is directly involved in merge-mining. This accumulated SHA256 hashrate results in more accumulated work, increasing the security against certain double-spend attacks. Also, the hashrate accumulated not only includes 100% of Bitcoin hashrate, but also hashrate from bitcoin forks.